src/rdkafka_lz4.c

```diff
@@ -259,20 +259,26 @@ rd_kafka_lz4_decompress (rd_kafka_broker_t *rkb, int proper_hc, int64_t Offset,
 
                 /* Need to grow output buffer, this shouldn't happen if
                  * contentSize was properly set. */
-                if (unlikely(r > 0 && out_of == outlen)) {
+                if (unlikely(out_of == outlen)) {
                         char *tmp;
-                        size_t extra = (r > 1024 ? r : 1024) * 2;
+                        // Grow exponentially with some factor > 1 (using 1.75)
+                        // for ammortized O(1) copying.
+                        size_t extra = outlen * 3 / 4;
```

Comment:
`LZ4F_decompress()` returns the number of bytes written for the next call, so we should still use `r` here.

Please retain the original factor of 2 and the min `1024` check unless there is clear motivation why it needs to be changed.

Reply:
Actually, it returns the number of *compressed* bytes it expects for the next call, not the number of *decompressed* bytes it guesses will return on the next call. See `lz4frame.h`:

```
 * @return is an hint of how many `srcSize` bytes LZ4F_decompress() expects for next call.
 * Schematically, it's the size of the current (or remaining) compressed block + header of next block.
 * Respecting the hint provides some small speed benefit, because it skips intermediate buffers.
 * This is just a hint though, it's always possible to provide any srcSize.
 * When a frame is fully decoded, @return will be 0 (no more data expected).
 * If decompression failed, @return is an error code, which can be tested using LZ4F_isError().
```

> Please retain the original factor of 2

I'm not a huge fan of a factor of 2 for the reasons described here: https://github.com/facebook/folly/blob/master/folly/docs/FBVector.md

Chiefly, any factor > 1 will ensure amortized O(1) copying behavior, but a factor of 2 is worst for memory fragmentation, because the memory allocator has no chance at reusing the previously allocated blocks.

> and the min `1024`

Sure, my mistake. I'd misread that as a a *max* of 1024 when I first read the code.